bib/cmd/bootc-image-builder/mtls.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/osbuild/images/pkg/rpmmd"
+	"github.com/sirupsen/logrus"
+)
+
+type mTLSConfig struct {
+	key  []byte
+	cert []byte
+	ca   []byte
+}
+
+type fileReader interface {
+	ReadFile(string) ([]byte, error)
+}
+
+func extractTLSKeys(reader fileReader, repoSets map[string][]rpmmd.RepoConfig) (*mTLSConfig, error) {
+	var keyPath, certPath, caPath string
+	for _, set := range repoSets {
+		for _, r := range set {
+			if r.SSLClientKey != "" {
+				if keyPath != "" && (keyPath != r.SSLClientKey || certPath != r.SSLClientCert || caPath != r.SSLCACert) {
+					return nil, fmt.Errorf("multiple TLS client keys found, this is currently unsupported")
+				}
+
+				keyPath = r.SSLClientKey
+				certPath = r.SSLClientCert
+				caPath = r.SSLCACert
+			}
+		}
+	}
+	if keyPath == "" {
+		return nil, nil
+	}
+
+	key, err := reader.ReadFile(keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read TLS client key from the container: %w", err)
+	}
+
+	cert, err := reader.ReadFile(certPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read TLS client certificate from the container: %w", err)
+	}
+
+	ca, err := reader.ReadFile(caPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read TLS CA certificate from the container: %w", err)
+	}
+
+	return &mTLSConfig{
+		key:  key,
+		cert: cert,
+		ca:   ca,
+	}, nil
+}
+
+// prepareOsbuildMTLSConfig writes the given mTLS keys to the given directory and returns the environment variables
+// to set for osbuild
```

Comment:
(super nitpick) it now also returns the cleanup that the user should call when done with the vars